utils/global-state-update-gen/src/generic/state_tracker.rs

```diff
@@ -63,6 +63,23 @@ impl<T: StateReader> StateTracker<T> {
         let _ = self.entries_to_write.insert(key, value);
     }
 
+    pub fn write_bid(&mut self, bid_kind: BidKind) {
+        let bid_addr = match bid_kind.clone() {
+            BidKind::Unified(bid) => BidAddr::from(bid.validator_public_key().clone()),
+            BidKind::Validator(validator_bid) => {
+                BidAddr::from(validator_bid.validator_public_key().clone())
+            }
+            BidKind::Delegator(delegator) => BidAddr::new_from_public_keys(
+                delegator.validator_public_key(),
+                Some(delegator.delegator_public_key()),
+            ),
+        };
+
+        let _ = self
+            .entries_to_write
+            .insert(bid_addr.into(), bid_kind.into());
+    }
```

Comment:
> I think this function should also update self.staking, so that it we never add an entry to entries_to_write without also reflecting the changes in the cache.

I see that you update self.staking separately in set_bid, but I think it would be safer to do this in this function. 

Are you positive about that? I believe I'm mirroring what the original code was doing in regards to bids_cache / staking and entries_to_write

Reply:
After discussion, Bart agreed with me on the main point, and I agreed with him on the secondary point.